globus_automate_client/cli/auth.py

```diff
@@ -295,32 +316,34 @@ def get_access_token_for_scope(scope: str) -> Optional[str]:
     return token
 
 
-def logout(token_store: str = DEFAULT_TOKEN_FILE) -> bool:
-    try:
-        os.remove(token_store)
-    except OSError:
-        click.echo("couldn't remove token cache file", err=True)
-        return False
+def logout(token_store: Union[pathlib.Path, str] = DEFAULT_TOKEN_FILE) -> bool:
+    cache = TokenCache(token_store)
+    cache.load_tokens()
+    cache.clear_tokens()
+    cache.save_tokens()
     return True
 
 
-def revoke_login(token_store: str = DEFAULT_TOKEN_FILE) -> bool:
+def revoke_login(token_store: Union[pathlib.Path, str] = DEFAULT_TOKEN_FILE) -> bool:
     client = _get_globus_sdk_native_client(CLIENT_ID, CLIENT_NAME)
     if not client:
         click.echo("failed to get auth client", err=True)
         return False
     cache = TokenCache(token_store)
     cache.load_tokens()
-    if not logout(token_store):
-        return False
-    for token_set in cache.tokens.values():
+
+    def revoker(scope: str, token_set: TokenSet) -> bool:
```

Comment:
I guess I could make a case that the revoke should be a method on the `TokenSet` itself (or on the token store) and the client would be a parameter to the operation, since presumable the `TokenSet` would know how to revoke its tokens. That being said I can also see how it's nice to keep the `TokenSet` kinda dumb and keep the logic elsewhere. Up to you if you want to move the revoke operation into the class.

Reply:
Yeah, that's a good point. Also a good point about keeping it dumb. I think of it as really managing the persistent store more than having any real understanding of Auth. I guess that breaks in that it has the update from the oauth response thing, so maybe that should be generic-ized to some other update sort of function, but I'll keep it as is for now.